src/ProposalTypesConfigurator.sol

```diff
@@ -64,38 +71,130 @@ contract ProposalTypesConfigurator is IProposalTypesConfigurator {
+            if (_proposalTypes[proposalTypeId].txTypeHashes[i] == txTypeHash) {
+                revert NoDuplicateTxTypes();
+            }
+        }
+
+        _proposalTypes[proposalTypeId].txTypeHashes.push(txTypeHash);
+    }
+
     /**
      * @notice Set the parameters for a proposal type. Only callable by the admin or timelock.
      * @param proposalTypeId Id of the proposal type
      * @param quorum Quorum percentage, scaled by `PERCENT_DIVISOR`
      * @param approvalThreshold Approval threshold percentage, scaled by `PERCENT_DIVISOR`
      * @param name Name of the proposal type
      * @param module Address of module that can only use this proposal type
+     * @param txTypeHashes A list of transaction function selectors that represent the type hash, i.e. keccak256("foobar(uint,address)")
      */
     function setProposalType(
         uint8 proposalTypeId,
         uint16 quorum,
         uint16 approvalThreshold,
         string calldata name,
-        address module
+        address module,
+        bytes32[] memory txTypeHashes
     ) external override onlyAdminOrTimelock {
-        _setProposalType(proposalTypeId, quorum, approvalThreshold, name, module);
+        _setProposalType(proposalTypeId, quorum, approvalThreshold, name, module, txTypeHashes);
     }
 
     function _setProposalType(
         uint8 proposalTypeId,
         uint16 quorum,
         uint16 approvalThreshold,
         string calldata name,
-        address module
+        address module,
+        bytes32[] memory txTypeHashes
     ) internal {
         if (quorum > PERCENT_DIVISOR) revert InvalidQuorum();
-        if (approvalThreshold > PERCENT_DIVISOR) {
-            revert InvalidApprovalThreshold();
-        }
+        if (approvalThreshold > PERCENT_DIVISOR) revert InvalidApprovalThreshold();
+
+        _proposalTypes[proposalTypeId] = ProposalType(quorum, approvalThreshold, name, module, txTypeHashes);
+        _proposalTypesExists[proposalTypeId] = true;
 
-        _proposalTypes[proposalTypeId] = ProposalType(quorum, approvalThreshold, name, module);
+        emit ProposalTypeSet(proposalTypeId, quorum, approvalThreshold, name, txTypeHashes);
+    }
 
-        emit ProposalTypeSet(proposalTypeId, quorum, approvalThreshold, name);
+    /**
+     * @notice Adds an additional scope for a given proposal type.
+     * @param proposalTypeId Id of the proposal type
+     * @param scope An object that contains the scope for a transaction type hash
+     */
+    function updateScopeForProposalType(uint8 proposalTypeId, Scope calldata scope) external override onlyAdmin {
+        _updateScopeForProposalType(proposalTypeId, scope);
+    }
+
+    /**
+     * @notice Adds an additional scope for a given proposal type.
+     * @param proposalTypeId Id of the proposal type
+     * @param scope An object that contains the scope for a transaction type hash
+     */
+    function _updateScopeForProposalType(uint8 proposalTypeId, Scope calldata scope) internal {
```

Comment:
we need to check for valid parameters in `scope`, like length of params and comparators